cpp/src/arrow/status.h

```diff
@@ -84,6 +84,7 @@ enum class StatusCode : char {
   PlasmaObjectNonexistent = 21,
   PlasmaStoreFull = 22,
   PlasmaObjectAlreadySealed = 23,
+  StillExecuting = 24
```

Comment:
Wasn't sure about this. We might call this a more generic "Not ready" status code